src/utilities.cpp

```diff
@@ -556,4 +559,132 @@ void findAllVariablesWithEquivalences(const ComponentPtr &component, VariablePtr
     }
 }
 
+IdList listIds(const ModelPtr &model, bool mathIds)
+{
+    // Collect all existing ids in a list and return. NB can't use a map or a set as we need to be able to print
+    // invalid models (with duplicated ids) too.
+
+    std::unordered_set<std::string> idList;
+    // Model.
+    std::string id = model->id();
+    if (!id.empty()) {
+        idList.insert(id);
+    }
+    // Units.
```

Comment:
doesn't there need to be something to pull IDs out of imports for imported units?

Reply:
You tell me :) From our earlier conversations I'd thought that IDs on imports were discarded? See the last "see more" block here: https://cellml-specification.readthedocs.io/en/latest/reference/formal_and_informative/specA2.html
Is the comment on the arrow in the example wrong? If so, I can change it there and add storage for imported ids here?

Reply:
And a followup question - if IDs **are** imported, and the requirement for their uniqueness is over the document, not the model, does the flattener check for this when merging document scopes? I need to check ...

Reply:
Ignore me ... I get it now :)

Reply:
Fixed in 45a1c42b